suit_generator/suit/manifest.py

```diff
@@ -212,6 +215,35 @@ class SuitComponentVersion(SuitList):
 
     _metadata = Metadata(children=[SuitInt])
 
+    @staticmethod
+    def _convert_version_part(part):
+        class PrereleaseType(Enum):
+            alpha = -3
+            beta = -2
+            rc = -1
+
+        if isinstance(part, str):
+            if part.isnumeric():
+                return int(part)
+            else:
+                try:
+                    prerelease = getattr(PrereleaseType, part)
+                    return prerelease.value
+                except AttributeError:
+                    raise ValueError(f"Unsupported prerelease type: {part}")
+        elif isinstance(part, int):
+            return part
+        else:
+            raise ValueError(f"Unsupported version part: {part}")
+
+    @classmethod
+    @log_call
+    def from_obj(cls, obj: Union[List[int], str]) -> SuitList:
+        """Restore SUIT representation from passed object."""
+        if isinstance(obj, str):
+            obj = [cls._convert_version_part(part) for part in obj.split(".")]
```

Comment:
So the string format would be for example `1.0.0.alpha` (meaning "." before the prerelease instead of for example "-" like `1.0.0-alpha`?
Perhaps both should be possible?

Reply:
Fixed it!